checks/branch_protection.go

```diff
@@ -153,115 +165,122 @@ func isBranchProtected(branches []*github.Branch, name string) (bool, error) {
+
+	score += requiresThoroughReviews(protection, branch, dl)
+
+	if protection.GetEnforceAdmins() != nil &&
+		protection.EnforceAdmins.Enabled {
+		dl.Info("'admininistrator' PRs need reviews before being merged on branch '%s'", branch)
+		score += 3
+	} else {
+		dl.Warn("'admininistrator' PRs are exempt from reviews on branch '%s'", branch)
 	}
 
-	if requiresThoroughReviews(protection, branch, dl) {
-		totalSuccess++
+	if score == totalScore {
+		return checker.MaxResultScore
 	}
 
-	return checker.CreateProportionalScoreResult(CheckBranchProtection,
-		"%d out of %d branch protection settings are enabled", totalSuccess, totalChecks)
+	return checker.CreateProportionalScore(score, totalScore)
 }
 
 // Returns true if several PR status checks requirements are enabled. Otherwise returns false and logs why it failed.
-func requiresStatusChecks(protection *github.Protection, branch string, dl checker.DetailLogger) bool {
-	// This is disabled by default (bad).
-	if protection.GetRequiredStatusChecks() != nil &&
-		protection.RequiredStatusChecks.Strict &&
-		len(protection.RequiredStatusChecks.Contexts) > 0 {
-		return true
+// Maximum score returned is 2.
+func requiresStatusChecks(protection *github.Protection, branch string, dl checker.DetailLogger) int {
+	score := 0
+
+	if protection.GetRequiredStatusChecks() == nil ||
+		!protection.RequiredStatusChecks.Strict {
+		dl.Warn("status checks for merging disabled on branch '%s'", branch)
+		return score
 	}
-	switch {
-	case protection.RequiredStatusChecks == nil ||
-		!protection.RequiredStatusChecks.Strict:
-		dl.Warn("Status checks for merging disabled on branch '%s'", branch)
-	case len(protection.RequiredStatusChecks.Contexts) == 0:
-		dl.Warn("Status checks for merging have no specific status to check on branch '%s'", branch)
-	default:
-		panic("Unhandled status checks error")
+
+	dl.Info("strict status check enabled on branch '%s'", branch)
+	score++
+
+	if len(protection.RequiredStatusChecks.Contexts) > 0 {
+		dl.Warn("status checks for merging have specific status to check on branch '%s'", branch)
+		score++
+	} else {
+		dl.Warn("status checks for merging have no specific status to check on branch '%s'", branch)
 	}
-	return false
+
+	return score
 }
 
 // Returns true if several PR review requirements are enabled. Otherwise returns false and logs why it failed.
-func requiresThoroughReviews(protection *github.Protection, branch string, dl checker.DetailLogger) bool {
-	// This is disabled by default (bad).
-	if protection.GetRequiredPullRequestReviews() != nil &&
-		protection.RequiredPullRequestReviews.RequiredApprovingReviewCount >= minReviews &&
-		protection.RequiredPullRequestReviews.DismissStaleReviews &&
-		protection.RequiredPullRequestReviews.RequireCodeOwnerReviews {
-		return true
-	}
-	if protection.RequiredPullRequestReviews == nil {
-		dl.Warn("Pullrequest reviews disabled on branch '%s'", branch)
-		return false
+// Maximum score returned is 5.
```

Comment:
i'm not good at real math but is it 7?

Reply:
you're right! I fiddles so much with the numbers and forgot to update the comment. Thanks!